src/app/forms/components/number-input/number-input.component.ts

```diff
@@ -15,10 +15,31 @@ import { BaseControlComponent } from '../base-control/base-control.component';
   ],
 })
 export class NumberInputComponent extends BaseControlComponent implements AfterContentInit {
+  @Input() vehicleType?: string | null;
   get style(): string {
     return `govuk-input ${this.width ? `govuk-input--width-${this.width}` : ''}`;
   }
 
+  get getWarningMessage(): string {
+
+    if (this.isCorrectVehicleType()) {
+      if (this.shouldDisplayLengthWarning()) return 'This length dimension field value is greater than 12,000mm. Check your input before proceeding';
+      if (this.shouldDisplayWidthWarning()) return 'This width dimension field value is greater than 2,600mm. Check your input before proceeding';
+    }
+    return '';
+  }
+
+  shouldDisplayLengthWarning(): boolean {
+    return this.label === 'Length' && Number(this.value) > 12000;
```

Comment:
This is fine for your use case, but I would caution against using Number for processing what I assume are string values as it might not always work as expected, e.g. for the opposition condition:

Number('') < 12000  // true
Number(null) < 12000 // true

Whereas parseInt might be more inline with the behaviour you actually want, e.g:

parseInt(null) < 12000 // false
parseInt('') < 12000 // false

Reply:
That is a good point Bardy! I will take more attention in this similar situation.

Reply:
```suggestion
    return this.label === 'Length' && parseInt(this.value) > 12000;
```